src/mlx/warnings.py

```diff
@@ -234,6 +239,9 @@ def warnings_wrapper(args):
             warnings.activate_checker_name('xmlrunner')
         if args.coverity:
             warnings.activate_checker_name('coverity')
+        if args.robot:
+            robot_checker = warnings.activate_checker_name('robot')
+            robot_checker.checkers = [RobotSuiteChecker(args.name, verbose=args.verbose)]
```

Comment:
ok this makes now sense why you return checker from `activate_checker_name`. This seems quite tricky though - should we not rather provide an class API to do this?

Reply:
I agree that it's tricky and should be improved. There is more code in this repo that is not pythonic in my opinion.